src/video/cocoa/SDL_cocoaopengl.m

```diff
@@ -135,7 +136,11 @@ - (void)explicitUpdate
     if ([NSThread isMainThread]) {
         [super update];
     } else {
-        dispatch_async(dispatch_get_main_queue(), ^{ [super update]; });
+        if (SDL_GetHintBoolean(SDL_HINT_MAC_OPENGL_SYNC_DISPATCH, SDL_FALSE)) {
```

Comment:
SDL_GetHint* are not optimized functions. For something like this I'd recommend checking this during initialization or setting a callback that sets a variable when the hint changes. Look at `SDL_GameControllerButtonReportingHintChanged()` for an example of the latter use case.

Reply:
Good note. Having a callback would probably be optimal to capture post-initialisation changes, will go with that.